### ISSUE TITLE
Getting an invalid context value warning

### ISSUE THREAD

**Issue by ccec1t91077** [getsentry/sentry-php#1526]
### Environment

sentry/sdk 3.3.0
sentry/sentry 3.13.1
sentry/sentry-laravel 3.2.0

php 8.2.5
laravel 10.0.3

### Steps to Reproduce

1. Add context

I'm try to add some info context to sentry by below code, when there is no context set, I'll pass an empty array.

```php
$this->reportable(function (Throwable $e) {
    if (!app()->environment(['local', 'testing'])) {
        if (method_exists($e, 'getExtraContext')) {
            \Sentry\configureScope(function (\Sentry\State\Scope $scope) use ($e) {
                $scope->setContext('[Extra Context]', $e->getExtraContext());
            });
        }
        Integration::captureUnhandledException($e);
    }
});
```

This is how I set the context, there will be some exception use this trait：

```php
namespace App\Traits;

trait HasExtraContext
{
    protected array $extraContext = [];

    public function setExtraContext(array $context): void
    {
        $this->extraContext = array_merge($this->extraContext, $context);
    }

    public function getExtraContext(): array
    {
        return $this->extraContext;
    }
}

```

3. View exceptions info from web backend

Getting warning on web backend.

```
There was 1 problem processing this event

contexts.[Extra Context]: Discarded invalid value
Reason: expected an object
Value: []
```

### Expected Result

According to the rules of `setContext()`, second parameter should be able to pass an array, but I'm getting warning if passed an empty array.

```php
/**
 * Sets context data with the given name.
 *
 * @param string               $name  The name that uniquely identifies the context
 * @param array<string, mixed> $value The value
 *
 * @return $this
 */
public function setContext(string $name, array $value): self
{
    $this->contexts[$name] = $value;

    return $this;
}
```

Seems there got something wrong. I expected that it should be an empty field on backend.

**Comment by ccec1t91077** [comment]
I'm not sure if this issue should be in this repository, because actually I'm using the `sentry/sentry-laravel` package on Laravel project. I just list to the environment by what I saw from `composer show | grep sentry`.

And, I'm coming from the Laravel documentation page link, so I'm sorry if the question was posted to a repository that it shouldn't belong to.

BTW, apart from the warning, the function working fine on exception info, also the `setContext()` working well if I'm not sending an empty array.

Just let you know I'm not hurry on this issue, because of it will not affect the normal process.

Thanks in advance.

**Comment by cleptric** [comment]
The Laravel SDK is built on top of the PHP SDK. The code affected lives in this repository, https://github.com/getsentry/sentry-php/blob/master/src/State/Scope.php#L120-L133.

We might need to add a safeguard around setting an empty array as a context value or lose our requirements on the protocol level.

**Comment by cleptric** [comment]
@ccec1t91077 This was fixed and released in [3.18.1](https://github.com/getsentry/sentry-php/releases/tag/3.18.1)
